lib/runner.go

```diff
@@ -10,43 +11,67 @@ import (
 	"path/filepath"
 	"regexp"
 	"sort"
-	"sync"
 	"syscall"
 	"time"
 )
 
-// Runner runs a series of tests and displays its results
+// Runner runs a series of tests and displays its results.
 type Runner struct {
 	// Flags
-	TestTargets []string
-	IncludeRe   string
-	ExcludeRe   string
-	Timeout     time.Duration
-	InOrder     bool
-	RandomSeed  int64
-	JSONOutput  bool
-	Verbose     bool
-	DryRun      bool
+	testTargets  []string
+	includeReStr string
+	excludeReStr string
+	timeout      time.Duration
+	inOrder      bool
+	randomSeed   int64
+	jsonOutput   bool
+	verbose      bool
+	dryRun       bool
 
 	testResults       []TestResult
 	failedTestResults []TestResult
 }
 
-// RunCommand is the public entrypoint of the Runner
-// It gathers test scripts, runs them, and displays the result
+// NewRunner constructs a new Runner.
+func NewRunner(
+	testTargets []string,
+	includeReStr string,
+	excludeReStr string,
+	timeout time.Duration,
+	inOrder bool,
+	randomSeed int64,
+	jsonOutput bool,
+	verbose bool,
+	dryRun bool,
+) *Runner {
```

Comment:
Too many arguments, I would personally avoid overloading `NewRunner` with all those args and would try using options. That way if you ever need to throw in more arguments you don't have to change the `NewRunner` signature. 
e.g `func NewRunner(options *Options) *Runner `

Reply:
I'm not a fan of service locators as they make it much harder to maintain breaking changes. Yes, we don't change the signature, but at the cost of hiding the underlying changes. I can see the advantages of this pattern, as it makes the reading more clear. I'll move them to an `options`-like struct, but I reckon it can become a mess.

Reply:
The pattern I suggested is not really a service locator as your use case is much more trivial than "encapsulating services". In terms of readability you are not sacrificing anything as the caller would be constructing an option as oppose to passing the individual parameters. The optional parameters are the recommended and more semantic way of doing this in golang. Dave Chaney does a very good job of explaining the advantage of optional parameters (although his use case is more complex but the idea is the same), I recommend checking it out.

Reply:
I'll check it out.